extension/entitystore/serviceprovider.go

```diff
@@ -206,12 +213,12 @@ func (s *serviceprovider) serviceAttributeFromIamRole() ServiceAttribute {
 }
 
 func (s *serviceprovider) serviceAttributeFromAsg() ServiceAttribute {
-	if s.ec2Info == nil || s.ec2Info.GetAutoScalingGroup() == "" {
+	if s.getAutoScalingGroup() == "" {
 		return ServiceAttribute{}
 	}
 
 	return ServiceAttribute{
-		Environment: "ec2:" + s.ec2Info.GetAutoScalingGroup(),
+		Environment: "ec2:" + s.autoScalingGroup,
```

Comment:
Should this line also call `s.getAutoScalingGroup()` to ensure we are thread-safe?

Reply:
Addressed in revision